org.dbgate.DbGate.yml

```diff
@@ -0,0 +1,71 @@
+app-id: org.dbgate.DbGate
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '23.08'
+command: dbgate
+rename-icon: dbgate
+finish-args:
+  - --device=dri
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --filesystem=home
+  - --talk-name=org.freedesktop.Notifications
+modules:
+  - name: dbgate
+    buildsystem: simple
+    build-commands:
+      - chmod +x dbgate.AppImage
+      - ./dbgate.AppImage --appimage-extract
+      - rm dbgate.AppImage
+
+      - desktop-file-edit --set-icon ${FLATPAK_ID} --set-key Exec --set-value 'dbgate
+        %u' squashfs-root/dbgate.desktop
+      - install -Dm644 squashfs-root/dbgate.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+
+      - install -Dm755 dbgate.sh ${FLATPAK_DEST}/bin/dbgate
+      - install -Dm644 org.dbgate.DbGate.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml
+
+      # - install -Dm644 squashfs-root/usr/share/mime/dbgate.xml ${FLATPAK_DEST}/share/mime/packages/${FLATPAK_ID}.xml
+      # - rm squashfs-root/usr/share/mime/dbgate.xml
+
+      - mkdir -p ${FLATPAK_DEST}/share/icons/hicolor
+      - cp -r squashfs-root/usr/share/icons/hicolor/* ${FLATPAK_DEST}/share/icons/hicolor
+      # flatpak doesn't support icons lareger than 512x512
+      - rm -rf ${FLATPAK_DEST}/share/icons/hicolor/1024x1024
+
+      - rm -rf squashfs-root/resources/public/icons/win squashfs-root/resources/public/icons/mac
+
+      - mv squashfs-root ${FLATPAK_DEST}/dbgate
+
+      # To allow separate locales
+      # https://searchfox.org/mozilla-central/rev/8a4f55bc09ffc5c25dcb4586c51ae4a9fee77b4c/taskcluster/docker/firefox-flatpak/runme.sh#131-133
+      - |
+        for lang in ${FLATPAK_DEST}/dbgate/locales/*.pak
+        do
+          locale="$(basename -s .pak $lang)"
+          install -Dm644 -t "${FLATPAK_DEST}/share/runtime/locale/${locale%%-*}/" "$lang"
+          ln -sf "${FLATPAK_DEST}/share/runtime/locale/${locale%%-*}/$(basename $lang)" "${FLATPAK_DEST}/dbgate/locales/$(basename $lang)"
+        done
+    sources:
+      - type: file
+        only-arches: [x86_64]
+        url: https://github.com/dbgate/dbgate/releases/download/v5.2.8-beta.22/dbgate-5.2.8-beta.22-linux_x86_64.AppImage
```

Comment:
why not building from source?

Reply:
Because network requests are prohibited while flatpak build. Prebuilt node modules are downloaded during the build and precaching is very difficult

Reply:
there is tooling for that

https://github.com/flatpak/flatpak-builder-tools

Reply:
I saw several electron apps on flathub and they use appimage too. Electron build process is quite complicated, especially when using natrive modules, so I prefer to have core build only on one place - that is github pipeline

Reply:
> I saw several electron apps on flathub and they use appimage too.

two wrongs don't make a right.

Reply:
Sorry, now is build from source for me too difficult. Also it could easily break, when two complicated and very different pipelines (github, flathub) must be maintained. It is possible to use appimage for now?